logsmanagement/parser.c

```diff
@@ -0,0 +1,1622 @@
+    wblp_config->num_fields = num_fields;
+    wblp_config->delimiter = delimiter;
+    
+    char **parsed_format = parse_csv(log_format, delimiter, num_fields); // parsed_format is NULL-terminated
+    wblp_config->fields = callocz(num_fields, sizeof(web_log_line_field_t));
+    unsigned int fields_off = 0;
+
+    for(int i = 0; i < num_fields; i++ ){
+
+        if(strcmp(parsed_format[i], "$host:$server_port") == 0 || 
+           strcmp(parsed_format[i], "%v:%p") == 0) {
+            wblp_config->fields[fields_off++] = VHOST_WITH_PORT;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$host") == 0 || 
+           strcmp(parsed_format[i], "$http_host") == 0 ||
+           strcmp(parsed_format[i], "%v") == 0) {
+            wblp_config->fields[fields_off++] = VHOST;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$server_port") == 0 || 
+           strcmp(parsed_format[i], "%p") == 0) {
+            wblp_config->fields[fields_off++] = PORT;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$scheme") == 0) {
+            wblp_config->fields[fields_off++] = REQ_SCHEME;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$remote_addr") == 0 || 
+           strcmp(parsed_format[i], "%a") == 0 ||
+           strcmp(parsed_format[i], "%h") == 0) {
+            wblp_config->fields[fields_off++] = REQ_CLIENT;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$request") == 0 || 
+           strcmp(parsed_format[i], "%r") == 0) {
+            wblp_config->fields[fields_off++] = REQ;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$request_method") == 0 || 
+           strcmp(parsed_format[i], "%m") == 0) {
+            wblp_config->fields[fields_off++] = REQ_METHOD;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$request_uri") == 0 || 
+           strcmp(parsed_format[i], "%U") == 0) {
+            wblp_config->fields[fields_off++] = REQ_URL;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$server_protocol") == 0 || 
+           strcmp(parsed_format[i], "%H") == 0) {
+            wblp_config->fields[fields_off++] = REQ_PROTO;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$request_length") == 0 || 
+           strcmp(parsed_format[i], "%I") == 0) {
+            wblp_config->fields[fields_off++] = REQ_SIZE;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$request_time") == 0 || 
+           strcmp(parsed_format[i], "%D") == 0) {
+            wblp_config->fields[fields_off++] = REQ_PROC_TIME;
+            continue;
+        }
+
+        if(strcmp(parsed_format[i], "$status") == 0 || 
+           strcmp(parsed_format[i], "%>s") == 0 ||
+           strcmp(parsed_format[i], "%s") == 0) {
+            wblp_config->fields[fields_off++] = RESP_CODE;
```

Comment:
`RESP_CODE` is associated with status here, is it not better to name it as `RESP_STATUS`. Or is it in a wrong place? :thinking:

Reply:
I think I followed again the approach that the Go plugin was written with, so any naming conventions should match the ones from there.